src/preferences/common/preference-service.spec.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { JsonPreferenceService, PreferenceService } from './preference-service';
+import { FileSystem, FileStat } from "../../filesystem/common/filesystem";
+import { FileSystemWatcher } from "../../filesystem/common/filesystem-watcher"
+import "mocha";
+import * as chai from "chai";
+import * as chaiAsPromised from "chai-as-promised";
+
+
+const expect = chai.expect;
+let prefService: PreferenceService;
+let fileWatcher: FileSystemWatcher;
+let fs: FileSystem;
+
+before(() => {
```

Comment:
I have just figured out today, it is an anti-pattern. And all of us copies the same code. before, and before each should go to the first describe scope. Otherwise, this `beforeEach` will be called before all test suites for all available *.spec.ts test module. I will write it down and share my findings with the whole team.